src/envoy/server/crud/doe.py

```diff
@@ -1,38 +1,59 @@
-from datetime import date, datetime, timedelta
+from datetime import datetime
 from typing import Optional, Sequence, Union
 
-from sqlalchemy import TIMESTAMP, Select, cast, func, select
+from sqlalchemy import Select, func, literal_column, select
 from sqlalchemy.ext.asyncio import AsyncSession
 
 from envoy.server.crud.common import localize_start_time, localize_start_time_for_entity
+from envoy.server.model.archive.doe import ArchiveDynamicOperatingEnvelope as ArchiveDOE
 from envoy.server.model.doe import DynamicOperatingEnvelope as DOE
 from envoy.server.model.site import Site
 
 
-async def select_doe_for_scope(
+async def select_doe_include_deleted(
     session: AsyncSession,
     aggregator_id: int,
-    site_id: Optional[int],
+    site_id: int,
     doe_id: int,
-) -> Optional[DOE]:
-    """Attempts to fetch a doe using its' DOE id, also scoping it to a particular aggregator/site
+) -> Optional[Union[DOE, ArchiveDOE]]:
+    """Attempts to fetch a doe using its' DOE id, also scoping it to a particular aggregator/site. The archive
+    table will also be checked for deleted instances (of which the most recent deletion will be matched).
 
     aggregator_id: The aggregator id to constrain the lookup to
-    site_id: If None - no effect otherwise the query will apply a filter on site_id using this value"""
-
-    stmt = (
-        select(DOE, Site.timezone_id)
-        .join(DOE.site)
-        .where((DOE.dynamic_operating_envelope_id == doe_id) & (Site.aggregator_id == aggregator_id))
-    )
-    if site_id is not None:
-        stmt = stmt.where(DOE.site_id == site_id)
+    site_id: the query will apply a filter on site_id using this value"""
 
-    resp = await session.execute(stmt)
-    raw = resp.one_or_none()
-    if raw is None:
+    # Start by confirming the referenced site_id exists within the specified aggregator.
```

Comment:
This first query is probably redundant, site entity would have already been pulled earlier as part of aggregator/site validation check - so just pass timezone from that

Reply:
Fixed in latest commit